src/pages/dish-order.jsx

```diff
@@ -1,74 +1,154 @@
 /* eslint-disable react/prop-types */
-import { Box, Modal } from "@mui/material"
+import { Box, Dialog, Modal, Tooltip } from "@mui/material"
 import close_white from '/imgs/close-white.svg'
 import close from '/imgs/close.svg'
 import DishOptions from "../cmps/dish-options"
 import { useState } from "react"
 import { useMediaQuery } from "react-responsive"
 import AppFooter from "../cmps/app-footer"
+import { useDispatch, useSelector } from "react-redux"
+import { ADD_TO_BAG, CLEAR_BAG } from "../store/restaurant/restaurant.reducer"
+import { useLocation } from "react-router"
+import question from '/imgs/question.svg'
+import { toggleBag } from "../store/restaurant/restaurant.actions"
 
-export default function DishOrder({ dish, toggleModal, isModalOpen }) {
+export default function DishOrder({ dish, toggleModal, isModalOpen, isOpenNow, restaurant }) {
     const [selectedOptions, setSelectedOptions] = useState({
         sideDish: '',
         changes: [],
         quantity: 1,
     })
+    const [isPopupOpen, setIsPopupOpen] = useState(false)
     const isMobile = useMediaQuery({ query: '(max-width: 769px)' })
+    const dispatch = useDispatch()
+    const location = useLocation()
+    const bag = useSelector(storeState => storeState.restaurantModule.bag)
+    const isDisabled = !selectedOptions.sideDish || !selectedOptions.quantity
+    const isrRestaurantPage = location.pathname.includes('/restaurant')
+
+    function onAddToBag() {
+        if (bag.length > 0 && bag[0].restaurantName !== restaurant.name) {
```

Comment:
You can write it like this:
   if (bag.length && bag[0].restaurantName !== restaurant.name)